### PR TITLE
MacOS builds on CI

### PR DESCRIPTION
**Pull request by bkase**
MacOS circle CI build with a setup script any macOS user can invoke.

### CODE REVIEW

File: scripts/macos-setup.sh [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  env TERM=xterm opam pin -y add src/external/ocaml-dune
+  eval $(opam config env)
+
+  # TODO: Change when Corey's PR lands
+  # install rocks fails trying to ldconfig
+  cd src/external/ocaml-rocksdb; ./install_rocksdb.sh || true; cd ../../../
+  sudo update_dyld_shared_cache
+
+  # All our ocaml packages
+  env TERM=xterm opam switch import src/opam.export
```
Comment (cmr) [inline scripts/macos-setup.sh]: macos terminals don't export TERM???

File: scripts/macos-setup.sh [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+# Compile things
+if [[ $COMPILE_THINGS == "YES" ]]; then
+  # All our ocaml packages
+  env TERM=xterm opam switch import src/opam.export
+  eval $(opam config env)
+
+  # Our pins
+  env TERM=xterm opam pin -y add src/external/ocaml-sodium
+  env TERM=xterm opam pin -y add src/external/ocaml-rocksdb
```
Comment (cmr) [inline scripts/macos-setup.sh]: remove this, we don't use opam for this anymore.

File: scripts/macos-setup.sh (hunk 1 above)
Comment (cmr) [reply scripts/macos-setup.sh]: oh i see

File: Makefile [hunk 3]
```diff
@@ -93,14 +92,25 @@ else
...
+
 ########################################
 ## Containers and container management
 
 
 # push steps require auth on docker hub
 docker-toolchain:
 	@if git diff-index --quiet HEAD ; then \
-		docker build --file dockerfiles/Dockerfile-toolchain --tag codaprotocol/coda:toolchain-$(GITLONGHASH) . && \
+		docker build --no-cache --file dockerfiles/Dockerfile-toolchain --tag codaprotocol/coda:toolchain-$(GITLONGHASH) . && \
```
Comment (cmr) [inline Makefile]: rip my container build times

Comment (bkase) [reply Makefile]: oh I didn't mean to do that

File: Makefile [hunk 4]
```diff
@@ -265,24 +275,17 @@ docs/res/%.dot.png: docs/res/%.dot
 	dot -Tpng $< > $@
 
 docs/res/%.tex.pdf: docs/res/%.tex
-	cd docs/res && pdflatex $(notdir $<)
+	cd docs/res && (pdflatex $(notdir $<) || lualatex $(notdir $<))
```
Comment (cmr) [inline Makefile]: did you intend to install lualatex with that script?

File: Makefile (hunk 3 above)
Comment (cmr) [reply Makefile]: I'm glad!

File: src/external/ocaml-rocksdb/dune [hunk 5]
```diff
@@ -32,7 +32,7 @@
       git fetch
       git clean -dxf
       git checkout v5.17.2
-      env ROCKSDB_DISABLE_SNAPPY=1 ROCKSDB_DISABLE_GFLAGS=1 ROCKSDB_DISABLE_LZ4=1 ROCKSDB_DISABLE_ZSTD=1 ROCKSDB_DISABLE_NUMA=1 ROCKSDB_DISABLE_TBB=1 ROCKSDB_DISABLE_JEMALLOC=1 ROCKSDB_DISABLE_TCMALLOC=1 ROCKSDB_DISABLE_BACKTRACE=1 make static_lib -j$(nproc)
+      env ROCKSDB_DISABLE_SNAPPY=1 ROCKSDB_DISABLE_GFLAGS=1 ROCKSDB_DISABLE_LZ4=1 ROCKSDB_DISABLE_ZSTD=1 ROCKSDB_DISABLE_NUMA=1 ROCKSDB_DISABLE_TBB=1 ROCKSDB_DISABLE_JEMALLOC=1 ROCKSDB_DISABLE_TCMALLOC=1 ROCKSDB_DISABLE_BACKTRACE=1 make static_lib -j2
```
Comment (cmr) [inline src/external/ocaml-rocksdb/dune]: suggestion:
```
case $(uname) in
'Linux') nproc=nproc ;;
'Darwin') nproc=sysctl -n hw.ncpu
*) echo I do not know how to build on $(uname) ; exit 1
esac
env .... make static_lib $($nproc)
```

File: scripts/macos-setup.sh [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+set -eu
+
+USAGE="$0 <download|opam|compile|all>"
+
+if [[ $# -eq 0 ]]; then
+  echo $USAGE
+  exit 1
+fi
+
+if [[ $1 == "download" ]]; then
```
Comment (cmr) [inline scripts/macos-setup.sh]: want to make this a `case`?

**Pull request merged**